File: C_Sharp/Monoid/AddMonoid.cs
using System;

namespace Monoid
{
    [Serializable]
    public class AddMonoid : IMonoid<int>
    {
        private static readonly AddMonoid instance = new AddMonoid();

        protected AddMonoid()
        {
        }

        public int NeutralElement
        {
            get { return 0; }
        }

        public int Operation(int fst, int snd)
        {
            return fst + snd;
        }

        public static AddMonoid Instance
        {
            get { return instance; }
        }
    }
}

File: C_plus_plus/AddMonoid.cpp
#pragma once
#include"AbstractMonoid.cpp"

class AddMonoid : public AbstractMonoid
{
public:
	
	int NeutralElement()
	{
		return 0;
	}

	int Operation(int fst, int snd)
	{
		return fst + snd;
	}
};